xbrowse_server/phenotips/views.py

```diff
@@ -1,90 +1,187 @@
+from xbrowse_server.phenotips.utilities import convert_internal_id_to_external_id
+from xbrowse_server.phenotips.utilities import get_uname_pwd_for_project
+from xbrowse_server.phenotips.utilities import get_generic_collaborator_uname_pwd_for_project
+
+from xbrowse_server.base.models import Project
+from django.shortcuts import render, redirect, get_object_or_404
+  
 logger = logging.getLogger(__name__)
 
-@log_request('sync_data_api')
+  
+@log_request('phenotips_proxy_edit_page')
 @login_required
-#handle a request to sync data between xbrowse and phenotips
-def process_sync_request(request):
-  '''handle a request to sync data'''
-  message={'message':'meant for AJAX POST'}
-  if request.method == 'POST':
-      if request.is_ajax():
-          uname = request.POST.get('uname')
-          pwd = request.POST.get('pwd')
-          individual_id=request.POST.get('id')
-          if __process_sync_request_helper(uname,pwd,individual_id):
-            message={'status':'success'}
-          else:
-            message={'status':'error'}
-  return JsonResponse(message)
+#test function to see if we can proxy phenotips
+#exempting csrf here since phenotips doesn't have this support
+@csrf_exempt
+def fetch_phenotips_edit_page(request,eid):
+  '''test function to see if we can proxy phenotips'''  
+  current_user = request.user
+  
+  if request.GET.has_key('project'):
+    project_name=request.GET['project']  
+    #add project name to session for use in proxying
+    request.session['current_project_name']=project_name
+    
+    #also put current ext_id into session object
+    phenotips_proj_specific_admin__uname,phenotips_proj_specific_admin_pwd = get_uname_pwd_for_project(project_name)
+    ext_id=convert_internal_id_to_external_id(eid,
+                                            phenotips_proj_specific_admin__uname,
+                                            phenotips_proj_specific_admin_pwd)
+    request.session['current_ext_id']=ext_id
+    
+    #no check current auth level and add that to session too
+    project = get_object_or_404(Project, project_id=project_name)
+    if project.can_admin(request.user):
+        auth_level = 'admin'
+    elif project.can_edit(request.user):
+        auth_level = 'editor'
+    elif project.is_public:
+        auth_level = 'public'
+    elif project.can_view(request.user):
+        auth_level = 'viewer'
+    else:
+        raise Exception("Authx - cannot authenticate:fetch_phenotips_edit_page ")
+    request.session['current_auth_level']=auth_level
+    
+  else: 
+    #getting admin account to translate eid to id
+    project_name = request.session['current_project_name']
+    ext_id=request.session['current_ext_id']
+    auth_level = request.session['current_auth_level']
+    
+  #depending on auth level, pick either the full edit username or the view-only username for this
+  #project to fetch page
+  phenotips_uname,phenotips_pwd ='_','_'
```

Comment:
Done and committed. These are initialization habits left over from java and forgetting Python scope is simpler!